### ISSUE TITLE
Issue with Sec-WebSocket-Protocol on WS middleware

### ISSUE THREAD

**Issue by vasicvuk** [ThreeMammals/Ocelot#639]
## Expected Behavior / New Feature

I wanted to add downstream WS to Active MQ Artemis STOMP.

I added reroute configuration.
Reroute configuration:
```
{
      "DownstreamPathTemplate":"/stomp",
      "DownstreamScheme":"ws",
      "DownstreamHostAndPorts":[
        {
          "Host":"broker",
          "Port":61623
        }
      ],
      "UpstreamPathTemplate":"/broker/stomp"
    }
```

It is expected to work like this but.

## Actual Behavior / Motivation for New Feature

Actual behavior is that i get in browser:
Error during WebSocket handshake: Sent non-empty 'Sec-WebSocket-Protocol' header but no response was received.

Response code is: 101

Headers that are sent are:
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
Sec-WebSocket-Key: hMNbcPOBCOP9t4S9Xb4C6g==
Sec-WebSocket-Protocol: v10.stomp, v11.stomp, v12.stomp
Sec-WebSocket-Version: 13

Headers received:
Sec-WebSocket-Accept: 2koEJ/JywLlfry8DM/eMQDZgJD8=

But there is no Sec-WebSocket-Protocol in response. 

I guess the issue is here:
https://github.com/ThreeMammals/Ocelot/blob/89f0cc786aa0895a9945eeea9a8ba6ad5ba21fb9/src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs#L36

If  Sec-WebSocket-Protocol is not redirected then it will not give a response of supported protocol

There is no error i log

## Specifications

  - Version: 12
  - Platform: Kastrel
  - Subsystem: Linux

**Comment by vasicvuk** [comment]
I've debugged a code a little and i found out that it is hanging at line:
```
await wsToDownstreamService.ConnectAsync(uri, CancellationToken.None);
```
I am not sure what is problem why it won't connect since. Maybe it is issue with ClientWebSocket. With nginx same works.

**Comment by vasicvuk** [comment]
I created a pull request solving this issue #642 .

Ocelot WebSockets Middleware didn't copy all the headers and protocols needed for communication to happen. I used  https://github.com/aspnet/Proxy/blob/3015029f5169b699dfa16bcf85cfc7ca0aaddae7/src/Microsoft.AspNetCore.Proxy/ProxyAdvancedExtensions.cs as a reference for implementation.

With this middleware i don't have an issue anymore.

**Comment by TomPallister** [comment]
@vasicvuk thanks for your efforts on this and the PR, I will review ASAP.

**Comment by vasicvuk** [comment]
@TomPallister  Thank you Tom for fast review :). I will close the issue since it is fixed and merged. Hope there will be release with the fix soon :)

**Comment by TomPallister** [comment]
Released in 12.0.1, should be on NuGet soon :)
